src/libraries/System.Linq/src/System/Linq/Single.cs

```diff
@@ -84,6 +84,9 @@ public static TSource Single<TSource>(this IEnumerable<TSource> source, Func<TSo
         }
 
         public static TSource? SingleOrDefault<TSource>(this IEnumerable<TSource> source)
+            => source.SingleOrDefault(default(TSource));
+
+        public static TSource SingleOrDefault<TSource>(this IEnumerable<TSource> source, TSource defaultValue)
```

Comment:
Similar to https://github.com/dotnet/runtime/pull/48886/files#r592482631, I think it would make sense to express the new method in terms of 
```csharp
public static TSource? SingleOrDefault<TSource>(this IEnumerable<TSource> source)
```
as defined in `main`.

Reply:
This has been done. Should I rewrite `Single()` to use `TryGetSingle()` as well? I could cut down on a lot of repeated code.
```cs
public static TSource Single<TSource>(this IEnumerable<TSource> source)
{
    var single = source.TryGetSingle(out bool found);
    if (found)
    {
        return single!;
    }
    else
    {
        ThrowHelper.ThrowNoElementsException();
        return default; // Make the compiler happy until a never type is implemented
    }
}
```

Reply:
Sounds good 👍